File: python/fizzbuzz.py
# Time -> 0(N) Space -> 0(N)
def fizzBuzz(n: int) -> list[str]:
    result: list[str] = []
    idx = 1
    while idx <= n:
        fizz = idx % 3
        buzz = idx % 5

        match fizz:
            case 0 if buzz == 0:
                result.append("FizzBuzz")
            case 0:
                result.append("Fizz")
            case _ if buzz == 0:
                result.append("Buzz")
            case _:
                result.append(str(idx))

        idx += 1

    return result


print(fizzBuzz(3))
print(fizzBuzz(5))
print(fizzBuzz(15))

File: go/FizzBuzz/FizzBuzz.go
package main

import "fmt"

// Time -> 0(N) Space -> 0(N)
func fizzBuzz(n int) []string {
	result := make([]string, 0)
	idx := 1
	for idx <= n {
		fizz := idx % 3
		buzz := idx % 5
		switch {
		case buzz == fizz && buzz == 0:
			result = append(result, "FizzBuzz")
		case fizz == 0:
			result = append(result, "Fizz")
		case buzz == 0:
			result = append(result, "Buzz")
		default:
			result = append(result, fmt.Sprintf("%d", idx))
		}

		idx++
	}

	return result
}

func main() {
	fmt.Println(fizzBuzz(3))
	fmt.Println(fizzBuzz(5))
	fmt.Println(fizzBuzz(15))
}
